bench/db_bench.ml

```diff
@@ -0,0 +1,329 @@
+(** Benchmarks *)
+
+open Common
+open Rresult
+
+let src = Logs.Src.create "db_bench"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let seed = 1
+
+let () = Random.init seed
+
+let key_size = 44
+
+(* the hash_size in pack_index*)
+let hash_size = 30
+
+let value_size = 13
+
+let nb_entries = 1_000_000
+
+(* in tezos the log_size is 500_000  *)
+let log_size = nb_entries / 5
+
+let ( // ) = Filename.concat
+
+module Context = Make_context (struct
+  let key_size = key_size
+
+  let hash_size = hash_size
+
+  let value_size = value_size
+end)
+
+let entry_size = key_size + value_size
+
+let random =
+  let rec loop i acc =
+    if i = nb_entries then acc
+    else
+      let k = Context.Key.v () in
+      let v = Context.Value.v () in
+      loop (i + 1) ((k, v) :: acc)
+  in
+  loop 0 []
+
+let sorted = List.sort (fun a b -> String.compare (fst a) (fst b)) random
+
+let print_results db f nb_entries =
+  let _, time = with_timer f in
+  let micros = time *. 1_000_000. in
+  let op = micros /. float_of_int nb_entries in
+  let mb = float_of_int (entry_size * nb_entries / 1_000_000) /. time in
+  Log.info (fun l ->
+      l "%s: %f micros/op; \t %f MB/s \t total time = %fs.%!" db op mb time)
+
+let write_amplif offset nb_entries =
+  let stats = Index_unix.get_stats () in
+  let ratio_bytes =
+    float_of_int stats.bytes_written /. float_of_int (entry_size * nb_entries)
+  in
+  let ratio_reads = float_of_int stats.nb_writes /. float_of_int nb_entries in
+  let ratio_file_size =
+    float_of_int offset /. float_of_int (entry_size * nb_entries)
+  in
+  Log.info (fun l ->
+      l
+        "\twrite amplification in bytes = %f; in nb of writes = %f; in \
+         file_size = %f%!"
+        ratio_bytes ratio_reads ratio_file_size)
+
+let read_amplif nb_entries =
+  let stats = Index_unix.get_stats () in
+  let ratio_bytes =
+    float_of_int stats.bytes_read /. float_of_int (entry_size * nb_entries)
+  in
+  let ratio_reads = float_of_int stats.nb_reads /. float_of_int nb_entries in
+  Log.info (fun l ->
```

Comment:
`Log.app` is probably better here to avoid cluttering the output with debug prefixes.